cni/installer/install.go

```diff
@@ -30,22 +30,74 @@ const (
 	defaultSrcDirLinux      = "/output/"
 	defaultBinDirLinux      = "/opt/cni/bin/"
 	defaultConflistDirLinux = "/etc/cni/net.d/"
+	defaultLogFile          = "/var/log/azure-vnet.log"
+	transparent             = "transparent"
+	bridge                  = "bridge"
 
 	envCNIOS                     = "CNI_OS"
 	envCNITYPE                   = "CNI_TYPE"
 	envCNISourceDir              = "CNI_SRC_DIR"
 	envCNIDestinationBinDir      = "CNI_DST_BIN_DIR"
 	envCNIDestinationConflistDir = "CNI_DST_CONFLIST_DIR"
 	envCNIIPAMType               = "CNI_IPAM_TYPE"
+	envCNIMode                   = "CNI_MODE"
 	envCNIExemptBins             = "CNI_EXCEMPT_BINS"
+	envCNILogFile                = "CNI_LOG_FILE"
 )
 
-type environmentalVariables struct {
+type installerConfig struct {
 	srcDir         string
 	dstBinDir      string
 	dstConflistDir string
 	ipamType       string
 	exemptBins     map[string]bool
+	logFile        string
+	osType         string
+	cniType        string
+	cniMode        string
+}
+
+func (i *installerConfig) SetExempt(exempt []string) {
+	// set exempt binaries to skip installing
+	// convert to all lower case, strip whitespace, and split on comma
+	for _, binName := range exempt {
+		i.exemptBins[binName] = true
+	}
+}
+
+func (i *installerConfig) SetOSType(osType string) error {
+	if strings.EqualFold(osType, linux) || strings.EqualFold(osType, windows) {
+		i.osType = fmt.Sprintf("%s_%s", osType, amd64)
+	} else {
+		return fmt.Errorf("No target OS type supplied, please set %q env and try again", envCNIOS)
+	}
+	return nil
+}
+
+func (i *installerConfig) SetCNIType(cniType string) error {
+	// get paths for singletenancy and multitenancy
+	switch {
+	case strings.EqualFold(cniType, multitenancy):
+		i.cniType = fmt.Sprintf("%s-%s", cni, multitenancy)
+	case strings.EqualFold(cniType, singletenancy):
+		i.cniType = cni
+	default:
+		return fmt.Errorf("No CNI type supplied, please set %q env to either %q or %q and try again", envCNITYPE, singletenancy, multitenancy)
+	}
+	return nil
+}
+
+func (i *installerConfig) SetCNIMode(cniMode string) error {
+	// get paths for singletenancy and multitenancy
+	if cniMode != "" {
```

Comment:
one more check if ipv6mode is set, then mode can be only bridge..can;t be transparent